openapi.yaml

```diff
@@ -0,0 +1,149 @@
+        base_volume: { type: number, format: double, description: "Transaction amount in base pair volume." }
+        target_volume: { type: number, format: double, description: "Transaction amount in target pair volume." }
+        trade_timestamp: { type: integer, description: "Unix timestamp in milliseconds for when the transaction occurred." }
+        type: { type: string, description: "Used to determine the type of the transaction that was completed. Buy: Identifies an ask that was removed from the order book. Sell: Identifies a bid that was removed from the order book." }
+    Trades:
+      type: object
+      properties:
+        buy : { type: array, items: {$ref: '#/components/schemas/Trade' }}
+        sell : { type: array, items: {$ref: '#/components/schemas/Trade'}}
+      example:
+        {
+          "buy": [{ "trade_id":1234567, "price":"50.1", "base_volume":"0.1", "target_volume":"1", "trade_timestamp":"1700050000", "type":"buy" }],
+          "sell": [{ "trade_id":1234567, "price":"50.1", "base_volume":"0.1", "target_volume":"1", "trade_timestamp":"1700050000", "type":"sell" }]
+        }
+    Price:
+      type: object
+      properties:
+        coin_a_close: { type: number, format: double, description: Coin A close }
+        coin_a_high: { type: number, format: double, description: Coin B high }
+        coin_a_low: { type: number, format: double, description: Coin A low }
+        coin_a_open: { type: number, format: double, description: Coin A open }
+        coin_a_volume: {type: number, format: double, description: Coin A volume }
+        coin_b_close: { type: number, format: double, description: Coin B close }
+        coin_b_high: { type: number, format: double, description: Coin A high }
+        coin_b_low: { type: number, format: double, description: Coin B low }
+        coin_b_open: { type: number, format: double, description: Coin B open}
+        coin_b_volume: { type: number, format: double, description: Coin B volume }
+        count: { type: integer, description: Candlestick size }
+        time_period_start: { type: string , description: Trade timestamp, format: "date", example: "1985-04-12T23:00:00.00Z" }
+        time_period_end: { type: string , description: Trade timestamp, format: "date", example: "1985-04-12T23:15:00.00Z" }
+    Prices:
+      type: array
+      items: {$ref: '#/components/schemas/Price'}
+security:
+  - ApiKeyAuth: [read, write]
+tags: [{name: markets}, {name: orders}, {name: trades}, {name: prices}, {name: settings}]
+externalDocs:
+ description: Bot API Documentation
+ url: https://www.bot-api-docs.geniusyield.co/
+servers:
+  - url: https://localhost:4000/
+  - url: https://bot.api.geniusyield.co/
+paths:
+  /settings:
+    get:
+      tags: [settings]
+      responses: { 200: {description: successful operation, content: {application/json: {schema: {$ref: '#/components/schemas/Settings' }}}}}
+  /tickers:
```

Comment:
The name `tickers` comes form the Coin Gecko API requirements.

I was a bit torn if we should could this resource: market, pairs, pools or trading pairs. Tickers is actually not bad, although I would prefer markets personally.

The problem is: The Coin Gecko API requirements also defintes some query params that are called ticker_id.

If we go with this, then we need to call this resource Tickers - so the API stays nice and clear. 

Otherwise we would have Markets, which have ticker_id. This does not sounds really nice and clean to me.

Reply:
We introduced a new resource, called assets. Assets can have a ticker, wich is pretty much the asset name.